usb: display power supply overload notifications

hnez opened the pull request:
Add warning on the LCD …

![lcd](https://github.com/linux-automation/tacd/assets/1273320/95288f6d-73df-466a-9217-cff40a23fbfc)

… and web interface if the LXA TAC' USB ports are overloaded.

![web](https://github.com/linux-automation/tacd/assets/1273320/7bbbd6ac-83f7-4c9a-ae22-425315c2f88a)

KarlK90 commented on src/usb_hub.rs:
```diff
@@ -135,6 +136,47 @@ const PORTS: &[(&str, &str)] = &[
+
+impl OverloadedPort {
+    fn from_currents(total: f32, port1: f32, port2: f32, port3: f32) -> Option<Self> {
+        let total_overload = total > MAX_TOTAL_CURRENT * CURRENT_MARGIN;
+        let port1_overload = port1 > MAX_PORT_CURRENT * CURRENT_MARGIN;
+        let port2_overload = port2 > MAX_PORT_CURRENT * CURRENT_MARGIN;
+        let port3_overload = port3 > MAX_PORT_CURRENT * CURRENT_MARGIN;
```
Nit: please extract `MAX_PORT_CURRENT * CURRENT_MARGING` and `MAX_TOTAL_CURRENT * CURRENT_MARGING` into constants as well and use them directly in the comparison e.g.:

```rust
if total > MAX_TOTAL_CURRENT_XYZ {
  Some(Self::Total)
// ...
```

hnez replied:
Done

KarlK90 replied:
Thanks, I find that nicer. If you want to go all the way you could remove the `let total_overload = xxx` bindings and just do the comparison in the if branches. Do this at your own judgement, it is a nit for me.

KarlK90 commented on src/ui/screens/usb_overload.rs:
```diff
@@ -0,0 +1,150 @@
+                ui_text_style,
+            )
+            .draw(target)
+            .unwrap();
```
Are these infallible calls? Otherwise an error here would bring down `tacd`?

hnez replied:
> Are these infallible calls

In our case indeed. See [this discussion](https://github.com/linux-automation/tacd/pull/32#discussion_r1335978221) for more details.
I think the `.unwrap()`s are actually the cleanest solution here right now, as we have quite a lot of these kinds of draw calls and (at least with the current framebuffer implementation) none of them can fail.

KarlK90 replied:
Thanks for the reply, my response in the mentioned discussion applies here as well.

hnez commented:
I've implemented the overcurrent warning level constants as @KarlK90 suggested and fixed some merge conflicts while rebasing on the current `main` branch,
Please re-approve and merge at will.

hnez commented:
Changes introduced by #27 necessitated some changes with regard to error handling of ADC readings, which I have just pushed:

https://github.com/linux-automation/tacd/blob/b38944c6c5e0edc30a99aa4b4fd0449acfbb2237/src/usb_hub.rs#L291-L304

In addition to that I've also implemented @KarlK90's suggestion for inlining the `let overload_*` bindings into the `if … else if …` chain.

hnez commented:
Hi @KarlK90 can I get a new approval? The previous one was invalidated by required changes in a rebase.

KarlK90 reviewed (approved):
@hnez done!
